core/src/main/java/com/hcl/ecomm/core/services/impl/ProductServiceImpl.java

```diff
@@ -0,0 +1,168 @@
+		LOG.info("Final Token Value is : " + finalToken);
+		httpGet.setHeader("Authorization", finalToken);
+
+		try {
+			CloseableHttpResponse httpResponse = httpClient.execute(httpGet);
+
+			if (httpResponse.getStatusLine().getStatusCode() == 200) {
+				responseStream = EntityUtils.toString(httpResponse.getEntity());
+				LOG.info("Product Response Json : " + responseStream);
+
+			} else {
+				responseStream = "Failed to fetch products from the store";
+				LOG.error("Failed to fetch products from the store");
+			}
+		} catch (Exception e) {
+			LOG.error(" getAllProductDetails method caught an exception" + e.getMessage());
+			e.printStackTrace();
+		}
+		productJsonArray = ProductUtility.fromStringToJsonArray(responseStream);
+		LOG.info("Json Array formed : " + productJsonArray);
+
+		return productJsonArray;
+	}
+
+	@Override
+	public List<String> getAllProductSkus(JsonArray productJson) {
+
+		ArrayList<String> productSkuList = new ArrayList<String>();
+
+		for (JsonElement jel : productJson) {
+			String sku = jel.getAsJsonObject().get("sku").getAsString();
+			productSkuList.add(sku);
+		}
+		return productSkuList;
+	}
+
+	@Override
+	public JsonArray getProductDetail(String sku) {
+
+		LOG.info("inside getProductDetail method");
+		String token = loginService.getToken();
+		String domainName = getDomainName();
+		String servicePath = getServicePath();
+		String searchFirstCriteriaField = getSearchCriteriaField();
+		String searchFirstCriteriaValue = getSearchCriteriaValue();
+		String searchSecondCriteriaField = "SKU";
+		String searchSecondCriteriaValue = sku;
+
+		String productUrl = scheme + "://" + domainName + servicePath
+				+ "?searchCriteria[filterGroups][0][filters][0][field]=" + searchFirstCriteriaField
+				+ "&searchCriteria[filterGroups][0][filters][0][value]=" + searchFirstCriteriaValue
+				+ "&searchCriteria[filterGroups][0][filters][1][field]=" + searchSecondCriteriaField
+				+ "&searchCriteria[filterGroups][0][filters][1][value]=" + searchSecondCriteriaValue;
+
+		CloseableHttpClient httpClient = HttpClients.createDefault();
+		HttpGet httpGet = new HttpGet(productUrl);
+
+		httpGet.setHeader("Content-Type", "application/json");
+		String bearerToken = "Bearer " + token;
+		String finalToken = bearerToken.replaceAll("\"", "");
+		LOG.info("Final Token Value is : " + finalToken);
+		httpGet.setHeader("Authorization", finalToken);
+
+		try {
+			CloseableHttpResponse httpResponse = httpClient.execute(httpGet);
+
+			if (httpResponse.getStatusLine().getStatusCode() == 200) {
+				responseStream = EntityUtils.toString(httpResponse.getEntity());
+				LOG.info("Product Response Json : " + responseStream);
+			} else {
+				responseStream = "Failed to fetch products from the store";
+				LOG.error("Failed to fetch products from the store");
+			}
+		} catch (Exception e) {
+			LOG.error(" getProductDetail method caught an exception" + e.getMessage());
+			e.printStackTrace();
```

Comment:
@nitheshkumar-hcl  Why you kept e.printStackTrace().? pl remove.

Reply:
Removed the stack trace